### PR TITLE
Understanding AES and RSA encryption algorithms

### LINKED ISSUE #2856: Understanding AES and RSA encryption algorithms
### Introduction paragraph (2-3 paragraphs):
Encryption is basically a way of enciphering data especially when its on transit to prevent unauthorized personnel from  gaining access to it. Encryption is always done with intent of allowing only the intended recipient of a piece of data to have access to the data.  It is a key feature in maintaining the confidentiality, integrity and availability of data in communication.

In this article I will dwelling on the two methods of achieving encryption otherwise referred to as encryption algorithms. I will be diving deep into  AES and RSA encryption algorithms, discussing the steps involved in achieving encryption through them and the different reasons why they are in the front lane in the data encryption initiative.

### Key takeaways: 
1. The reader will be in a position to understand encryption as an emerging issue in information security and the ways of achieving encryption.
2. The reader will be able learn matters security through data encryption
3. The reader will be able to learn the two algorithms of achieving data security, that is AES and RSA.
### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by awinomackrine**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

This fixes #2856

### CODE REVIEW

**Comment by Dawe-7** [comment]
@hectorkambow, I would like to review this article.

**Comment by hectorkambow** [comment]
@awinomackrine 
Good afternoon and thank you for submitting your article for publishing.
Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀

Thank you.

File: content/articles/aes-rsa-encryption/index.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+5. [Getting started with RSA encryption algorithm](#getting-started-with-rsa-encryption-algorithm)
+6. [RSA algorithm functionality](#rsa-algorithm-functionality)
+7. [Conclusion](#conclusion)
+
+### Prerequisites
+
+For a reader to understand this article, they should have the following prior knowledge:
+
+- A basic understanding of the concept of data encryption and the need for encryption in data communication.
+- A beginner's understanding of data cryptography.
```
Comment (Bayler) [inline content/articles/aes-rsa-encryption/index.md:17]: You've written an article on data cryptography before, right? Please link it where you need the reader to have an understanding of data cryptography

File: content/articles/aes-rsa-encryption/index.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+As we already know, the encryption and decryption processes are done using a key.
+This block cipher uses a key that is 128 bits long for encryption and decryption of a message.
+- **AES-192**
+
+The AES-192 uses a key that is 192 bits long for the encryption and decryption of messages.
+
+- **AES-256**
+
+The AES-256 uses a key that is 256 bits long for the encryption and decryption of messages.
```
Comment (Bayler) [inline content/articles/aes-rsa-encryption/index.md]: Add a sentence or two more that add to the differences between these three block ciphers. I'm sure the number of bits isn't the only difference

File: content/articles/aes-rsa-encryption/index.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+Below is a simple diagrammatic representation flow of how AES encrypts data:
+
+![AES algorithm flow ](/engineering-education/aes-rsa-encryption/AES_flow.png)
+
+This illustrates that either of the 3 encryption keys is used to encrypt the message. The encryption process occurs in the "cipher" and an encrypted message is produced in the "ciphertext".
+
+The encryption process has 10 rounds for a 128-bit long key, 12 rounds for a 192-bit long key, and 14 rounds for a 256-bit long key.
+
+### Step 1: Generation of round keys
+
```
Comment (Bayler) [inline content/articles/aes-rsa-encryption/index.md:61]: This next steps illustrates......

You need to add a sentence in between the sentence and the header to guide the reader on what's happening in the next step. The transition needs to be smooth.

File: content/articles/aes-rsa-encryption/index.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+ - Send the public key (z,w) generated above to the person who wants to send a message and remain with the private key.
+ - The message sender turns the message into a number "G". The number G should be smaller than "z". This is done using an agreed-upon protocol commonly referred to as [Padding scheme](https://simple.wikipedia.org/wiki/RSA_algorithm#Padding_schemes)
+ - The sender then computes the ciphertext of the message using the expression, c = G^e  mod z where 'c' represents the ciphertext.
+
+The receiver can decrypt the message using the expression G = c^d  mod n according to the [chinese remainder theorem](https://simple.wikipedia.org/wiki/Chinese_remainder_theorem).
+
+### Conclusion
+
+Encryption is key in matters of information security, and it is vital to learn the different algorithms used in data encryption other than AES and RSA to broaden your horizons as a person in the computing world where cybercrime is increasing by the hour.
```
Comment (Bayler) [inline content/articles/aes-rsa-encryption/index.md:138]: Avoid the phrases "In conclusion" and "In summary". Phrases like these usually come across as cliche'. Also, your conclusion statement is too long. Consider breaking it into shorter sentences.

**Comment by Bayler** [comment]
@awinomackrine This was a nice read. I've seen the quality of your article has greatly improved from the last time I reviewed your article. Keep up!

Just some few corrections then we can move on to final review.

**Comment by awinomackrine** [comment]
@Bayler The additions have been made as suggested......Please check

**Comment by Bayler** [comment]
> @Bayler The additions have been made as suggested......Please check

@awinomackrine Are the images used in the article your own?

**Comment by awinomackrine** [comment]
@Bayler Not all, some are borrowed

**Comment by Bayler** [comment]
> @Bayler Not all, some are borrowed

@awinomackrine For those that are borrowed you have to add their sources

**Comment by awinomackrine** [comment]
> > @Bayler Not all, some are borrowed
> 
> @awinomackrine For those that are borrowed you have to add their sources

@Bayler done

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
Overall great work @awinomackrine 

cc @Bayler

**Pull request merged**